src/robotFactory.js

```diff
@@ -1,10 +1,110 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+    },
+    chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (this.coords.x === undefined) {
+      this.coords.x = 0;
+    }
```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```
I'm not sure if checking for undefined for coords x and y is necessary here as though we assign default values in property assignment.